ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx

```diff
@@ -550,12 +552,20 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
         authoringBase,
         formatMessage,
         modelIdByPath,
-        enqueueSnackbar
+        enqueueSnackbar,
+        env
       } = upToDateRefs.current;
       const { type, payload } = action;
       switch (type) {
         case guestSiteLoad.type:
         case guestCheckIn.type:
+          const { version: guestVersion } = payload;
+          const studioVersion = env.version;
+
+          if (type === guestCheckIn.type && (nou(guestVersion) || guestVersion.substr(0, 5) !== studioVersion)) {
```

Comment:
Where is the check against a list of valid versions (or ranges)? Did I miss it?

Reply:
@sumerjabri it will be checking 'base' versions, so if XB is for example `4.0.3.2` and studio is `4.0.3` it will match `4.0.3`, if not, it is considered outdated

Reply:
I don't like this. It should be checked against a list (or ranges) of compatible versions.

Reply:
This code is not meant to be forever. This is meant as a "_walk_" and on our next release, implement something more robust and comprehensive, test it appropriately and better understand what the checklist should be.

Note this logic is **not** embedded in client apps, it's in Studio so anyone who upgrades gets the update — even if their app embeds the SDK.

There is no list to be checked against just yet, and we don't know what the next version that breaks compatibility will be. Right now, is the first time we're implementing XB reporting its version and Studio checking it, hence, we're going to get either nothing (if the package is older), or `4.0.3` or eventually `4.0.3.{something}` if there's a patch. By the time there's a `4.0.4`, it means there's a new version of CrafterCMS out too with new improved compatibility checking logic. Makes sense?

If you insist, let's meet and decide on what the list or range will be. I don't have full grasp on what that should be which is why I wanted a _walk_ start as we approach `4.0.3` release.

Reply:
Reviewed with Sumer. Will start with this approach.